File: app/src/main/java/la/xiong/androidquick/GankApis.java
package la.xiong.androidquick;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;

/**
 * @author  ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
public interface GankApis {

    @GET("day/history")
    Observable<GankRes<List<String>>> getHistoryDate();

}

File: app/src/main/java/com/androidwind/androidquick/sample/network/GankApis.kt
package com.androidwind.androidquick.sample.network

import io.reactivex.Observable
import kotlinx.coroutines.Deferred
import retrofit2.http.GET

/**
 * @author  ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
interface GankApis {
    @GET("day/history")
    fun getHistoryDate(): Observable<GankRes<List<String>>>

    @GET("https://api.bintray.com/packages/ddnosh/maven/androidquick/images/download.svg")
    fun getSdkVersion(): Observable<String>

    @GET("day/history")
    fun getHistoryDateAsync(): Deferred<GankRes<List<String>>>

    @GET("day/h")
    fun getError(): Observable<GankRes<List<String>>>
}